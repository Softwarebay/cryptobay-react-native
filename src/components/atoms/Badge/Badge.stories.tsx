import { number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { Badge } from './Badge';

storiesOf('Atoms/Badge', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Green Badge', () => <Badge value={number('value', 4.3)} />)
  .add('Red Badge', () => <Badge value={number('value', -1.3)} />);
