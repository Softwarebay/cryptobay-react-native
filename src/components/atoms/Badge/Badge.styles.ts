import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  badge: {
    fontFamily: 'Poppins',
    fontSize: 14,
    lineHeight: 24,
    fontWeight: '600',
  },
  green: {
    color: '#76E268',
  },
  red: {
    color: '#EA3943',
  },
});
