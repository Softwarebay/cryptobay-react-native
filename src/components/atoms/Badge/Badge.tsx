import React from 'react';
import { Text, TextStyle } from 'react-native';

import { styles } from './Badge.styles';

type BadgeProps = {
  value: number;
  style?: TextStyle | TextStyle[];
  testID?: string;
};

export const Badge: React.FC<BadgeProps> = ({
  value,
  style,
  testID,
}: BadgeProps) => {
  const variant = value >= 0 ? 'green' : 'red';
  const handlePositiveValuePlus = value >= 0 && '+';

  return (
    <Text testID={testID} style={[styles.badge, styles[variant], style]}>
      {handlePositiveValuePlus}
      {value}%
    </Text>
  );
};
