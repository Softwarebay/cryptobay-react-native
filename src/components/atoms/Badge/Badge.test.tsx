import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Badge } from './Badge';

describe('Badge Atom', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(<Badge value={4.8} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure variants works correctly', () => {
    const { rerender } = render(<Badge value={4.8} testID="atom-badge" />);
    const badge = screen.getByTestId('atom-badge');
    expect(badge).toHaveTextContent('+4.8%');
    expect(badge).toHaveStyle({ color: '#76E268' });
    rerender(<Badge value={-1.3} testID="atom-badge" />);
    expect(badge).toHaveTextContent('-1.3%');
    expect(badge).toHaveStyle({ color: '#EA3943' });
  });
});
