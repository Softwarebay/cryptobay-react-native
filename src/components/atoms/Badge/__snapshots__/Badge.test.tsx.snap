// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Atom Component Snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Poppins",
        "fontSize": 14,
        "fontWeight": "600",
        "lineHeight": 24,
      },
      Object {
        "color": "#76E268",
      },
      undefined,
    ]
  }
>
  +
  4.8
  %
</Text>
`;
